test/functional/test_framework/util.py

```diff
@@ -32,6 +32,12 @@ def assert_approx(v, vexp, vspan=0.00001):
     if v > vexp + vspan:
         raise AssertionError("%s > [%s..%s]" % (str(v), str(vexp - vspan), str(vexp + vspan)))
 
+def assert_decimal_places(amount, decimal_places=8):
+    """Assert an amount has n digits (default 8 for amounts) after the decimal."""
+    remainder = str(amount).split('.')[-1]
+    if remainder != '0E-8':
```

Comment:
What cases are covered by this condition?

Reply:
The condition is for the case that there's no need to run the assert if the remainder string is the Python shorthand for the correct number of zero digits. But good catch: I should pass the param into it: s/`'0E-8'`/`'0E-{}'.format(decimal_places)`'

Reply:
Fixed and improved. Thanks @hebasto.